algorithms/kernel/service_heap.h

```diff
@@ -106,6 +131,50 @@ DAAL_FORCEINLINE void sortMaxHeap(RandomAccessIterator first, RandomAccessIterat
     }
 }
 
+template <CpuType cpu, typename RandomAccessIterator>
+DAAL_FORCEINLINE void sortMaxHeap(RandomAccessIterator first, RandomAccessIterator last)
+{
+    using Type            = typename RemovePointer<cpu, RandomAccessIterator>::type;
+    const auto comparator = [](const Type & a, const Type & b) -> bool { return a < b; };
+    sortMaxHeap<cpu>(first, last, comparator);
+}
+
+template <CpuType cpu, typename RandomAccessIterator, typename Addr>
+DAAL_FORCEINLINE void maxHeapUpdate(RandomAccessIterator first, Addr & prev, const Addr i)
```

Comment:
What is the reason for the function?

Reply:
No answer

Reply:
This function is called at least two times, so I think it is reasonable. Of course, suggesting it can be inlined properly.